### ISSUE TITLE
Twilio nuget package causes exceptions

### ISSUE THREAD

**Issue by davidpricedev** [digitalcoyote/NuGetDefense#65]
**Describe the bug**
`Twilio` nuget package breaks the checking process.

**To Reproduce**
1. Fire up a fresh, new, blank .NET framework console app project
2. Add Twilio `Install-Package Twilio` (version 5.55.0)
3. Add NuGetDefense `Install-Package NuGetDefense` (version 2.1.2)
4. Try building

** Actual behaviour **
Build fails with NuGetDefense errors unrelated to vulnerabilities: 

1>------ Rebuild All started: Project: ConsoleApp1, Configuration: Debug Any CPU ------
1>  ConsoleApp1 -> C:\Users\davidp\source\repos\ConsoleApp1\ConsoleApp1\bin\Debug\ConsoleApp1.exe
1>C:\Users\davidp\source\repos\ConsoleApp1\ConsoleApp1\packages.config : warning : Transitive dependency checking skipped
1>. 'dotnet list package --include-transitive' only supports SDK style NuGet Package References
1>C:\Users\davidp\source\repos\ConsoleApp1\ConsoleApp1\packages.config : error : NuGetDefense : NVD scan failed with exception: System.ArgumentException: '[7.x-1.1]' is not a valid version string.
1>     at NuGet.Versioning.VersionRange.Parse(String value, Boolean allowFloating)
1>     at NuGet.Versioning.VersionRange.Parse(String value)
1>     at NuGetDefense.NVD.Scanner.<>c__DisplayClass9_0.<GetVulnerabilitiesForPackages>b__1(String v)
1>     at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
1>     at NuGetDefense.NVD.Scanner.<>c__DisplayClass9_1.<GetVulnerabilitiesForPackages>b__0(String cve)
1>     at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
1>     at NuGetDefense.NVD.Scanner.GetVulnerabilitiesForPackages(NuGetPackage[] pkgs, Dictionary`2 vulnDict)

**Expected behavior**
Build succeeds or NuGetDefense generates errors indicating what vulnerabilities were found.

**Tools (please complete the following information):**
 - IDE: VS 2019
 - OS: Windows 10 20H2 19042.804
 - .net framework - 4.7.2

**Additional context**
It would be nice to at least make that error message more understandable and point to the source of the problem. Right now, I have only guesses as to whether this is a problem with one or more CVEs or a problem with Twilio's nuget package or a bug in NuGetDefense. 

**Existing Workarounds**
It does seem that I can work around this issue by ignoring Twilio via the same mechanism that NuGetDefense ignores itself. In other words, in the config file: 
```json
{
  "ErrorSettings": {
    "IgnoredPackages":  [     
      {
        "Id": "NugetDefense"
      },
      {
        "Id": "Twilio"
      }
    ],
  }
}
```

**Comment by digitalcoyote** [comment]
That looks like a version string that will need to be manually corrected. I'll try to get this in the next update. 

Note: It needs to validate version strings when building the NVD source as well (🪲).

**Comment by digitalcoyote** [comment]
Should be fixed in v1.0.17.1 and v2.1.2.1
